### DISCUSSION TITLE
Master/Worker服务注册时只能注册Hostname过去么？

### DISCUSSION THREAD

**Discussion by BongBongBang** [apache/dolphinscheduler#6699] [Q&A]
如上所属，如果只能注册hostname，那就是说，集群部署时，必须要配置hosts。

**Comment by BongBongBang** [comment]
可以注册ip过去么？

**Comment by BongBongBang** [comment]
我这里是在k8s部署时遇到的问题。因为没用helm安装，感觉那样不了解内容运行机理，所以，我是自己制作镜像安装的。经过查看dolphinscheduler的common项目下的NetUtil类getHost等方法的实现（zookeeper注册时，woker/master用来获取注册地址的首先），
通过给k8s#pod部署时指定hostname跟subdomain的方式，与K8s内部DNS访问方式配对，实现master/worker间的通讯

**Reply by ocean-zhc** [reply]
> 我这里是在k8s部署时遇到的问题。因为没用helm安装，感觉那样不了解内容运行机理，所以，我是自己制作镜像安装的。经过查看dolphinscheduler的common项目下的NetUtil类getHost等方法的实现（zookeeper注册时，woker/master用来获取注册地址的首先）， 通过给k8s#pod部署时指定hostname跟subdomain的方式，与K8s内部DNS访问方式配对，实现master/worker间的通讯

Hello, I also built the image by myself, but there is a problem, have you encountered it?
您好，我也是通过自己制作镜像构建的，但是有个问题，请教下是否有遇到过？

I started a discussion, please take a look: [https://github.com/apache/dolphinscheduler/discussions/11159](url)

**Comment by davidzollo** [comment]
English is needed

**Comment by huotianyu-CAa** [comment]
解决了我的问题，配置zookeeper集群时用的ip，结果dolpin总是zookeeper连接超时，改成hostname之后就好了
